main.go

```diff
@@ -0,0 +1,532 @@
+		fmt.Println()
+		log.Infof("Running build")
+
+		var isWorkspace bool
+		if xcworkspace.IsWorkspace(cfg.ProjectPath) {
+			isWorkspace = true
+		} else if !xcodeproj.IsXcodeProj(cfg.ProjectPath) {
+			failf("Project file extension should be .xcodeproj or .xcworkspace, but got: %s", filepath.Ext(cfg.ProjectPath))
+		}
+
+		// Build for simulator command
+		xcodeBuildCmd := xcodebuild.NewCommandBuilder(cfg.ProjectPath, isWorkspace, xcodebuild.BuildAction)
+		xcodeBuildCmd.SetScheme(cfg.Scheme)
+		xcodeBuildCmd.SetConfiguration(cfg.Configuration)
+
+		// Set simulator destination and disable code signing for the build
+		xcodeBuildCmd.SetCustomOptions([]string{"-destination", "id=" + simulatorID, "PROVISIONING_PROFILE_SPECIFIER="})
+
+		// Clean build
+		if cfg.IsCleanBuild {
+			xcodeBuildCmd.SetCustomBuildAction("clean")
+		}
+
+		// XcodeBuild Options
+		if cfg.XcodebuildOptions != "" {
+			options, err := shellquote.Split(cfg.XcodebuildOptions)
+			if err != nil {
+				failf("Failed to shell split XcodebuildOptions (%s), error: %s", cfg.XcodebuildOptions)
+			}
+			xcodeBuildCmd.SetCustomOptions(options)
+		}
+
+		// Output tool
+		{
+			if cfg.OutputTool == "xcpretty" {
+				xcprettyCmd := xcpretty.New(xcodeBuildCmd)
+
+				util.LogWithTimestamp(colorstring.Green, "$ %s", xcprettyCmd.PrintableCmd())
+				fmt.Println()
+
+				if rawXcodebuildOut, err := xcprettyCmd.Run(); err != nil {
+					log.Errorf("\nLast lines of the Xcode's build log:")
+					fmt.Println(stringutil.LastNLines(rawXcodebuildOut, 10))
+
+					if err := utils.ExportOutputFileContent(rawXcodebuildOut, rawXcodebuildOutputLogPath, bitriseXcodeRawResultTextEnvKey); err != nil {
+						log.Warnf("Failed to export %s, error: %s", bitriseXcodeRawResultTextEnvKey, err)
+					} else {
+						log.Warnf(`You can find the last couple of lines of Xcode's build log above, but the full log is also available in the raw-xcodebuild-output.log
+	The log file is stored in $BITRISE_DEPLOY_DIR, and its full path is available in the $BITRISE_XCODE_RAW_RESULT_TEXT_PATH environment variable
+	(value: %s)`, rawXcodebuildOutputLogPath)
+					}
+
+					failf("Build failed, error: %s", err)
+				}
+			} else {
+				util.LogWithTimestamp(colorstring.Green, "$ %s", xcodeBuildCmd.PrintableCmd())
+				fmt.Println()
+
+				buildRootCmd := xcodeBuildCmd.Command()
+				buildRootCmd.SetStdout(os.Stdout)
+				buildRootCmd.SetStderr(os.Stderr)
+
+				if err := buildRootCmd.Run(); err != nil {
+					failf("Build failed, error: %s", err)
+				}
+			}
+		}
+	}
+
+	//
+	// Export artifacts
+	var exportedArtifacts []string
+	{
+		fmt.Println()
+		log.Infof("Copy artifacts from Derived Data to %s", absOutputDir)
+
+		proj, _, err := findBuiltProject(cfg.ProjectPath, cfg.Scheme, cfg.Configuration)
+		if err != nil {
+			failf("Failed to open xcproj - (%s), error:", cfg.ProjectPath, err)
+		}
+
+		schemeBuildDir, err := buildTargetDirForScheme(proj, cfg.ProjectPath, cfg.Scheme, cfg.Configuration, cfg.SimulatorPlatform)
```

Comment:
buildTargetDirForScheme and exportArtifacts have the same inputs and lots of inputs, does it worth to split these functions?

Reply:
I think it is easier to read the code if they are separated.